rfcs/0020-security-on-call.md

```diff
@@ -0,0 +1,165 @@
+
+Members of this team should range from new contributors looking to
+participate, to more skilled and well known contributors.
+
+### Requirements
+
+Each team member should:
+
+ - Know how patches work, or at least be willing to learn
+ - Be comfortable with Git and our backporting workflow
+ - Know their personal limits and be confident asking for help
+
+## On Call
+
+An on-call rotation system should be used or made to handle scheduling
+and informing people about their shift. A shift is 24hrs long, and
+should probably start at midnight UTC, to be equally unfair to
+everyone.
+
+### Responsibilities
+
+1. Monitor a well defined list of mailing lists for new issues.
+2. Ensure each issue is triaged and addressed if needed.
+
+#2 is a bit vaguely worded, as the person is not required to
+_actually_ fix the issue. They are allowed to delegate the patching to
+other people. However, they _are_ responsible for ensuring the issue
+is _fixed_.
+
+#### Triage and Fixing
+
+1. Check to see if the issue impacts each supported version of NixOS.
+2. Write and / or backport patches as applicable, either by version
+   bumps, large patches, or minimal patches.
+3. Prepare an advisory to send to the nix-security-announce mailing
+   list, which a member of the NixOS Security Team will send.
+
+## The Well Defined List of Mailing Lists
+
+The list should not live in the RFC documentation, but an external set
+of documentation used to document the security patching process.
+However, an initial starting list to consider:
+
+1. oss-security
+2. full-disclosure
+3. an assortment of distro advisory announcements:
+
+ - Arch
+ - Debian
+ - Gentoo
+ - Red Hat
+ - SUSE
+
+## Ensuring Complete Mailing List Coverage
+
+This is a tricky problem, and I propose that the first implemention
+be naive, simple, quick, and ugly.
+
+I propose we have a shared email account with a norm that if you mark
+an issue read, you are obligated to handle the issue. Once issues are
+patched and released to channels, they should be removed from the
+inbox. Each member of the patching team will have access to the
+account.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This process will take time away from other projects contributors may
+be interested in undertaking.
+
+This project will introduce more mass rebuilds and additional load on
+Hydra.
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. RequestTracker for email-to-issues, but the RT module is somewhat
+   broken, not to mention the scars we all have around RT.
+2. Custom email-to-issue software
+3. Allow certain community members to be single points of failure
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+1. A place to house documentation and run-books for the patching team
+2. A review process for advisories
+3. Guidelines for backporting vs. separate patches when fixing a
+   package for Stable
+4. A tool for handling the On Call schedule asignments and
```

Comment:
All team members should be on a shared channel (mailing list, IRC, wire.com, Hangouts, whatever), and at the end of their shift the currently responsible team member should ping their successor and remind them about the upcoming shift.